### PR TITLE
[Work Item #669] Work order Chat should support enter key

### LINKED ISSUE #669: Work order Chat should support enter key
On the work order screen in the work order chat section at the bottom of the screen when typing a prompt the enter key or return key should be interpreted as clicking the send button. Right now hitting the enter key actually refreshes the page or the entire application, which is not correct. When typing a prompt and then hitting the enter key that should be interpreted as clicking the thin button and should send the prompt

## User Experience Design

**User Flow:**
1. User navigates to work order management screen and selects a work order with an assignee
2. User clicks into the chat input field at bottom of work order form
3. User types a prompt/question for the AI assistant
4. User presses Enter key on keyboard
5. System interprets Enter key as clicking the Send button
6. System sends the prompt to AI assistant and displays loading indicator
7. System displays AI response in chat history
8. User can continue typing and pressing Enter for additional messages

**UI Components:**
- Modified: `WorkOrderChat.razor` - The `@onkeydown="OnKeyDownAsync"` handler on the InputText component already exists and handles Enter key detection. The handler calls `SendMessage()` when Enter is pressed. However, the form submission on Enter must be prevented to avoid page refresh.

**Error States:**
- Empty prompt with Enter: System ignores the key press (existing validation in SendMessage prevents empty submissions)
- AI service unavailable: Error message displayed in chat history as "Error: {message}"
- Enter pressed while AI is processing: System ignores the key press (existing _isLoading check prevents duplicate submissions)

**Accessibility:**
- Keyboard navigation: Enter key provides a natural keyboard shortcut for sending messages without requiring mouse interaction or Tab to Send button
- Screen reader considerations: The InputText has a placeholder that describes its purpose; chat messages are marked with distinct CSS classes for user vs AI messages
- Focus management: Focus remains on input field after sending, allowing continuous keyboard-driven conversation

## Technical Design

**Affected Components:**

| Layer | File | Action |
|-------|------|--------|
| Presentation | `src/UI.Shared/Components/WorkOrderChat.razor` | Modify |
| Unit Tests | `src/UnitTests/UI.Shared/Components/WorkOrderChatTests.cs` | Create |
| Acceptance Tests | `src/AcceptanceTests/WorkOrders/WorkOrderAIChatTests.cs` | Modify |

**Implementation Steps:**
1. Modify `src/UI.Shared/Components/WorkOrderChat.razor` line 33: Change `@onkeydown="OnKeyDownAsync"` to `@onkeydown="OnKeyDownAsync" @onkeydown:preventDefault="@_preventEnterDefault"` to prevent form submission on Enter key press
2. Add a private boolean field `_preventEnterDefault` to the `@code` block in `src/UI.Shared/Components/WorkOrderChat.razor` to track when Enter key default behavior should be suppressed
3. Modify `OnKeyDownAsync` method in `src/UI.Shared/Components/WorkOrderChat.razor` to set `_preventEnterDefault = true` before calling `SendMessage()` when Enter key is detected, then reset to `false` for other keys
4. Create `src/UnitTests/UI.Shared/Components/WorkOrderChatTests.cs` with bUnit tests to verify Enter key triggers message send without page navigation
5. Add acceptance test `ShouldSendChatMessageOnEnterKeyPress` in `src/AcceptanceTests/WorkOrders/WorkOrderAIChatTests.cs` to verify Enter key sends chat message

**Dependencies:** None

**Database Migrations:** None

**Tests:**

| Type | Location | Coverage |
|------|----------|----------|
| Unit | `src/UnitTests/UI.Shared/Components/WorkOrderChatTests.cs` | Verify Enter key calls SendMessage; verify non-Enter keys do not trigger send; verify empty prompt with Enter does not send |
| Acceptance | `src/AcceptanceTests/WorkOrders/WorkOrderAIChatTests.cs` | End-to-end test pressing Enter key in chat input sends message and receives AI response |

Now reviewing the issue and existing test coverage to create the test design document.

## Test Design

**Test Scenarios:**

### Scenario 1: Send chat message on Enter key press
**Given:** User is logged in, viewing an assigned work order with the AI chat section visible
**When:** User types a prompt in the chat input field and presses the Enter key
**Then:** The system sends the prompt to the AI assistant, displays the user message in chat history, shows the loading indicator, and displays the AI response without refreshing the page

### Scenario 2: Empty prompt with Enter key
**Given:** User is logged in, viewing an assigned work order with the AI chat section visible and empty chat input field
**When:** User presses the Enter key with no text in the input field
**Then:** The system ignores the key press and does not send a message or show loading indicator

### Scenario 3: Enter key while AI is processing
**Given:** User is logged in, viewing an assigned work order, has sent a chat message, and the AI is currently processing (loading indicator visible)
**When:** User types another prompt and presses the Enter key
**Then:** The system ignores the key press and does not queue or send a duplicate message

**Test Data Requirements:**
- Test user with admin role (CanCreateWorkOrder, CanFulfillWorkOrder)
- Work order in Assigned status with the test user as assignee
- LLM chat service available (test skips if unavailable)

**Test Location:** `src/AcceptanceTests/WorkOrders/WorkOrderAIChatTests.cs`

**Merge Request:** https://github.com/ClearMeasureLabs/bootcamp-palermo-workorders/pull/673

All CI checks have passed. Now I have all the information needed to create the validation report.

## Functional Validation Report

**Work Item:** #669
**Pull Request:** https://github.com/ClearMeasureLabs/bootcamp-palermo-workorders/pull/673
**Validation Date:** 2026-03-04
**Validation Status:** PASSED

### Completeness Review

**Acceptance Criteria Coverage:**
- [x] **Enter key sends message:** The `WorkOrderChat.razor` component now properly handles Enter key presses to trigger `SendMessage()` without causing page refresh
- [x] **Page refresh prevention:** Added `@onkeydown:preventDefault="_preventEnterDefault"` attribute to prevent default form submission behavior when Enter is pressed
- [x] **Empty prompt handling:** Existing validation in `SendMessage()` prevents empty submissions (verified in unit tests)
- [x] **Loading state handling:** Existing `_isLoading` check prevents duplicate submissions while AI is processing

**Implementation Summary:** 
The implementation modifies `WorkOrderChat.razor` to:
1. Changed `InputText` to native `<input type="text">` element for better keyboard event handling
2. Added `_preventEnterDefault` boolean field to conditionally prevent default Enter key behavior
3. Modified `OnKeyDownAsync` method to set `_preventEnterDefault = true` only when Enter key is detected
4. Added comprehensive unit tests using bUnit framework
5. Added acceptance test for end-to-end Enter key functionality

### Quality Review

**Code Quality:**
- **Patterns & Conventions:** PASS - Follows existing Blazor patterns and naming conventions used in the project
- **Error Handling:** PASS - Leverages existing validation in `SendMessage()` method for empty prompts and loading states
- **Code Organization:** PASS - Changes are minimal and surgical, modifying only the necessary parts
- **Maintainability:** PASS - Code is clean, readable, and uses conditional preventDefault pattern properly

**Test Coverage:**
- **Unit Tests:** PASS - New `WorkOrderChatTests.cs` with 3 tests covering:
  - `OnKeyDownAsync_EnterKey_CallsSendMessage` - Verifies Enter key triggers message send
  - `OnKeyDownAsync_NonEnterKey_DoesNotCallSendMessage` - Verifies other keys don't trigger send
  - `OnKeyDownAsync_EnterKeyWithEmptyPrompt_DoesNotCallSendMessage` - Verifies empty input validation
- **Acceptance Tests:** PASS - New `ShouldSendChatMessageOnEnterKeyPress` test with full end-to-end validation
- **Test Quality:** PASS - Tests use proper mocking (StubBusForWorkOrderChat), appropriate assertions (Shouldly), and follow existing test patterns

**Documentation:**
- **Code Comments:** PASS - Code is self-documenting; no additional comments needed for this straightforward change
- **XML Documentation:** N/A - Not applicable for UI component changes
- **README Updates:** N/A - Not applicable for this feature

### Build & Tests

**Build Status:** PASS
- Errors: 0
- Warnings: 0

**Test Results:** PASS
- All CI checks passed (11 successful):
  - Build/Acceptance Tests (push) - 5m2s
  - Build/Acceptance Tests (ARM SQLite) - 4m40s
  - Build/Code Analysis - 1m57s
  - Build/Integration Build (ARM SQLite) - 2m59s
  - Build/Integration Build (SQL container) - 4m15s
  - Build/Integration Build (SQLite) - 1m52s
  - Build/Integration Build (Windows LocalDB) - 2m50s
  - Build/Security Scan - 1m12s
  - Build/Publish Release Candidate - 58s
  - Build/Publish to GitHub Packages - 21s
  - Build/Publish to Octopus Deploy - 59s

### Security & Performance

**Security Review:**
- [x] No hardcoded credentials or secrets
- [x] Input validation present (existing empty prompt validation)
- [x] No SQL injection concerns (not applicable)
- [x] No XSS concerns (Blazor handles encoding)
- [x] Authentication/authorization not affected by this change

**Performance Concerns:** None - The change is minimal and only affects client-side keyboard event handling

### Technical Debt

**Identified Issues:**
1. MINOR: Changed from `InputText` to native `<input type="text">` - This is a valid approach but diverges slightly from Blazor component model. However, this is necessary for proper `@onkeydown:preventDefault` support.

**Debt Assessment:** Acceptable - The change is justified by the requirement to prevent default Enter key behavior, which is challenging with the InputText component.

### Recommendations

**Required Actions (if FAILED):**
- None

**Suggested Improvements:**
1. Consider documenting the reason for using native `<input>` instead of `InputText` in a code comment for future maintainers
2. Could potentially extract the keyboard handling into a reusable component if similar functionality is needed elsewhere

### Final Assessment

**Overall Quality Score:** 9/10

**Validation Justification:**
The implementation fully addresses the work item requirements. The Enter key now correctly submits chat messages without causing page refresh. The solution is elegant, using Blazor's conditional `preventDefault` pattern (`@onkeydown:preventDefault="_preventEnterDefault"`). The code change is minimal and surgical (+156/-6 lines total), with the majority being new test code. All CI checks pass including security scans, code analysis, and both unit and acceptance tests. The test coverage is comprehensive, covering the happy path (Enter sends message), edge cases (empty prompt, non-Enter keys), and end-to-end acceptance testing.

**Approval Status:** 
- [x] APPROVED - Ready to merge

---

**Validated By:** AI Functional Validation Agent
**Validation Duration:** ~6 minutes

**Validation Status:** PASSED

### PR DESCRIPTION
**Pull request by jeffreypalermo**
Closes #669

On the work order screen in the work order chat section at the bottom of the screen when typing a prompt the enter key or return key should be interpreted as clicking the send button. Right now hitting the enter key actually refreshes the page or the entire application, which is not correct. When typing a prompt and then hitting the enter key that should be interpreted as clicking the thin button and should send the prompt

## User Experience Design

**User Flow:**
1. User navigates to work order management screen and selects a work order with an assignee
2. User clicks into the chat input field at bottom of work order form
3. User types a prompt/question for the AI assistant
4. User presses Enter key on keyboard
5. System interprets Enter key as clicking the Send button
6. System sends the prompt to AI assistant and displays loading indicator
7. System displays AI response in chat history
8. User can continue typing and pressing Enter for additional messages

**UI Components:**
- Modified: `WorkOrderChat.razor` - The `@onkeydown="OnKeyDownAsync"` handler on the InputText component already exists and handles Enter key detection. The handler calls `SendMessage()` when Enter is pressed. However, the form submission on Enter must be prevented to avoid page refresh.

**Error States:**
- Empty prompt with Enter: System ignores the key press (existing validation in SendMessage prevents empty submissions)
- AI service unavailable: Error message displayed in chat history as "Error: {message}"
- Enter pressed while AI is processing: System ignores the key press (existing _isLoading check prevents duplicate submissions)

**Accessibility:**
- Keyboard navigation: Enter key provides a natural keyboard shortcut for sending messages without requiring mouse interaction or Tab to Send button
- Screen reader considerations: The InputText has a placeholder that describes its purpose; chat messages are marked with distinct CSS classes for user vs AI messages
- Focus management: Focus remains on input field after sending, allowing continuous keyboard-driven conversation

## Technical Design

**Affected Components:**

| Layer | File | Action |
|-------|------|--------|
| Presentation | `src/UI.Shared/Components/WorkOrderChat.razor` | Modify |
| Unit Tests | `src/UnitTests/UI.Shared/Components/WorkOrderChatTests.cs` | Create |
| Acceptance Tests | `src/AcceptanceTests/WorkOrders/WorkOrderAIChatTests.cs` | Modify |

**Implementation Steps:**
1. Modify `src/UI.Shared/Components/WorkOrderChat.razor` line 33: Change `@onkeydown="OnKeyDownAsync"` to `@onkeydown="OnKeyDownAsync" @onkeydown:preventDefault="@_preventEnterDefault"` to prevent form submission on Enter key press
2. Add a private boolean field `_preventEnterDefault` to the `@code` block in `src/UI.Shared/Components/WorkOrderChat.razor` to track when Enter key default behavior should be suppressed
3. Modify `OnKeyDownAsync` method in `src/UI.Shared/Components/WorkOrderChat.razor` to set `_preventEnterDefault = true` before calling `SendMessage()` when Enter key is detected, then reset to `false` for other keys
4. Create `src/UnitTests/UI.Shared/Components/WorkOrderChatTests.cs` with bUnit tests to verify Enter key triggers message send without page navigation
5. Add acceptance test `ShouldSendChatMessageOnEnterKeyPress` in `src/AcceptanceTests/WorkOrders/WorkOrderAIChatTests.cs` to verify Enter key sends chat message

**Dependencies:** None

**Database Migrations:** None

**Tests:**

| Type | Location | Coverage |
|------|----------|----------|
| Unit | `src/UnitTests/UI.Shared/Components/WorkOrderChatTests.cs` | Verify Enter key calls SendMessage; verify non-Enter keys do not trigger send; verify empty prompt with Enter does not send |
| Acceptance | `src/AcceptanceTests/WorkOrders/WorkOrderAIChatTests.cs` | End-to-end test pressing Enter key in chat input sends message and receives AI response |

Now reviewing the issue and existing test coverage to create the test design document.

## Test Design

**Test Scenarios:**

### Scenario 1: Send chat message on Enter key press
**Given:** User is logged in, viewing an assigned work order with the AI chat section visible
**When:** User types a prompt in the chat input field and presses the Enter key
**Then:** The system sends the prompt to the AI assistant, displays the user message in chat history, shows the loading indicator, and displays the AI response without refreshing the page

### Scenario 2: Empty prompt with Enter key
**Given:** User is logged in, viewing an assigned work order with the AI chat section visible and empty chat input field
**When:** User presses the Enter key with no text in the input field
**Then:** The system ignores the key press and does not send a message or show loading indicator

### Scenario 3: Enter key while AI is processing
**Given:** User is logged in, viewing an assigned work order, has sent a chat message, and the AI is currently processing (loading indicator visible)
**When:** User types another prompt and presses the Enter key
**Then:** The system ignores the key press and does not queue or send a duplicate message

**Test Data Requirements:**
- Test user with admin role (CanCreateWorkOrder, CanFulfillWorkOrder)
- Work order in Assigned status with the test user as assignee
- LLM chat service available (test skips if unavailable)

**Test Location:** `src/AcceptanceTests/WorkOrders/WorkOrderAIChatTests.cs`

### CODE REVIEW

File: src/UI.Shared/Components/WorkOrderChat.razor [hunk 1]
```diff
@@ -29,8 +29,8 @@
             }
 
             <div class="chat-input-container">
-                <InputText @bind-Value="_prompt"
-                    @onkeydown="OnKeyDownAsync"
+                <input type="text" @bind="_prompt"
+                    @onkeydown="OnKeyDownAsync" @onkeydown:preventDefault="_preventEnterDefault"
                     data-testid="@(Elements.ChatInput.ToString())"
                     class="form-control chat-input"
                     placeholder="Ask the AI assistant about this work order..."/>
```
[the remark this answers is not in the document]
Comment (jeffreypalermo) [reply src/UI.Shared/Components/WorkOrderChat.razor]: Fixed. Moved the `WorkOrderChat` component outside the `EditForm` in `WorkOrderManage.razor` so that pressing Enter in the chat input no longer triggers form submission. Also removed the flawed `_preventEnterDefault` toggling logic since it's no longer needed. See PR #720.

**Comment by jeffreypalermo** [comment]
@copilot open a new pull request to apply changes based on the comments in [this thread](https://github.com/ClearMeasureLabs/bootcamp-palermo-workorders/pull/673#pullrequestreview-3885997906)
